security/advancedtls/advancedtls.go

```diff
@@ -492,7 +492,13 @@ func buildVerifyFunc(c *advancedTLSCreds,
 			}
 			// Perform default hostname check if specified.
 			if c.isClient && c.vType == CertAndHostVerification && serverName != "" {
-				opts.DNSName = serverName
+				parsedName, _, err := net.SplitHostPort(serverName)
```

Comment:
Do we have a test for this scenario? A positive and negative test.

Reply:
I added some tests... Default hostname check is a bit tricky to test since we need certs with hostname as CN and SAN... but yeah I added those test certs.

Reply:
Thanks.

Reply:
Great! Thanks for the review!